src/ape/managers/project.py

```diff
@@ -0,0 +1,1599 @@
+
+        else:
+            # Install all project's.
+            for item in self.config.dependencies:
+                dependency = self.add(item)
+                dependency.install(use_cache=use_cache)
+
+    def unpack(self, path: Path):
+        """
+        Move dependencies into a .cache folder.
+        Ideal for sandbox-projects.
+        """
+        cache_folder = path / ".cache"
+        for dependency in self.specified:
+            dependency.unpack(cache_folder)
+
+
+def load_manifest(path: Union[Path, str]) -> PackageManifest:
+    path = Path(path)
+    return (
+        PackageManifest.model_validate_json(path.read_text())
+        if path.is_file()
+        else PackageManifest()
+    )
+
+
+class ProjectManager(ExtraAttributesMixin, BaseManager):
+    """
+    The root project manager in Ape that can also create other projects.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if cls is ProjectManager:
+            # Using `ape.Project(path)`.
+            return super(cls, LocalProject).__new__(LocalProject)
+
+        elif len(args) >= 1 and isinstance(args[0], PackageManifest):
+            # Using `ape.Project.from_manifest()`.
+            return super(ProjectManager, Project).__new__(Project)
+
+        else:
+            # Using LocalProject.__init__ (internal).
+            return super(ProjectManager, LocalProject).__new__(LocalProject)
+
+    def __repr__(self) -> str:
+        try:
+            return repr(self._project)
+        except Exception:
+            # Exceptions not allowed in repr.
+            return "<ProjectManager>"
+
+    @only_raise_attribute_error
+    def __getattr__(self, name: str) -> Any:
+        return get_attribute_with_extras(self, name)
+
+    def __ape_extra_attributes__(self) -> Iterator[ExtraModelAttributes]:
+        # NOTE: This causes attributes to use the singleton local project.
+        #   However, you can still create additional projects via the
+        #   factory mixin methods.
+        yield ExtraModelAttributes(name="contracts", attributes=lambda: self._project)
+
+    @cached_property
+    def _project(self) -> "LocalProject":
+        return LocalProject(self.path)
+
+    @classmethod
+    def from_manifest(
+        cls, manifest: Union[PackageManifest, Path, str], config_override: Optional[Dict] = None
+    ) -> "Project":
+        """
+        Create an Ape project using only a manifest.
+
+        Args:
+            manifest (Union[PackageManifest, Path, str]): Either a manifest or a
+              path to a manifest file.
+            config_override (Optional[Dict]): Optionally provide a config override.
+
+        Returns:
+            :class:`~ape.managers.project.ProjectManifest`
+        """
+        return Project.from_manifest(manifest, config_override=config_override)
+
+    @classmethod
+    @contextmanager
+    def create_sandbox(cls, config_override: Optional[Dict] = None) -> Iterator["LocalProject"]:
```

Comment:
Suggestion: Not sure if `sandbox` is accurately descriptive, something containing `temporary` may be better

Reply:
Maybe `isolate()` or something? I want to also convey the existing project is copied into this temporary environment.

Reply:
`isolate` is good!

Was looking for something more commonly understood than `sandbox`

Something that denotes the action a bit better:
```py
with project.isolated_clone() as project:
    ...
```

Reply:
Here is the commonly understood concept of a sandbox: https://en.m.wikipedia.org/wiki/Sandbox_(software_development)

The reason I agree with you is because "sandbox" may imply some sort of testing or staging environment, whereas in our case it is supposed to be an isolated compilation environment, and those are not necessarily the same thing.

I like sandboxes too though because "they wash away" in the metaphor, like a temp folder.

Reply:
Come to think of it, however... the user could use this to test project changes without committing them..
But that isnt the reason it was made.

Hmmmmm